### ISSUE TITLE
Collection bindings do not update properly when Bindings.Update() is called

### ISSUE THREAD

**Issue by mikernet** [unoplatform/uno#7493]
### Current behavior

If the list instance stays the same then updating bindings does not update the items in the combo box:

```cs
public List<string> Items { get; set; } = new();

// on load or when page is navigation to:

var items = await GetItemsAsync();
Items.Clear();
Items.AddRange(items);
Bindings.Update();
```

```xml
<ComboBox ItemsSource="{x:Bind Items}"/>
```
Because the list instance is the same, the items are never updated. If you set the list to a new instance instead, the items are updated:
```cs
var items = await GetItemsAsync();
Items = new List<string>(items);
Bindings.Update();
```

### Works on UWP/WinUI

Yes

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia, Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### NuGet package version(s)

Probably every version but definitely all since 3.11.0-dev.210 including 4.x-dev.latest

### Affected platforms

iOS, Android, WebAssembly

**Comment by jeromelaban** [comment]
In general, you'll want to use an `ObservableCollection` instead of relying on `Bindings.Update()` to refresh the contents.

In this case, in UWP, the `ComboBox` content updates regardless of the use of `Bindings.Update()`, and is linked to the `ComboBox` enumerating the `ItemsSource` content when opening.

I'm curious though, where did you see this pattern being used ?

**Comment by mikernet** [comment]
Yeah, there are workarounds, but I was posting this because behavior does not have parity with UWP, which I presumed was a goal. 

The suggested workaround would require me to write a whole VM structure to map an existing structure that I don't have source access to modify. Would be nice if it just worked like UWP so I didn't have to mess with what already works.

**Comment by jeromelaban** [comment]
Thanks. This is still an unstable pattern in UWP/WinUI, and crashes at runtime.

You can reproduce it by clicking on the `ComboBox` before the `Items.Add(new SomeObject("Test")`, then opening it after the list has changed.

**Comment by mikernet** [comment]
The ComboBox can't be clicked in this case before that, it is disabled until that happens.

**Comment by MartinZikmund** [comment]
Still relevant for Skia targets

**Comment by MartinZikmund** [comment]
# Repro generator

This issue was classified as not reproducible with a code sample or test.

**Reason:** Collection binding update bug with Bindings.Update() — requires XAML compiled bindings (x:Bind) and cannot be tested via pure C# runtime test.
